urlf91495fd27_Rembau_complie_tgz-pJ8-matcher_MatchJ8/src/matcher/Match.java

```diff
@@ -0,0 +1,254 @@
+package matcher;
+
+import java.util.Hashtable;
+import java.util.LinkedList;
+import java.util.Stack;
+
+import NFAToDNA.Dtran;
+import NFAToDNA.NewNode;
+import NFAToDNA.ToDFA;
+import regexpToNFA.MainClass;
+import regexpToNFA.ManagerK;
+import regexpToNFA.Syntax;
+
+public class Match {
+	Hashtable<Integer,LinkedList<Dtran>> dtrans = new Hashtable<Integer,LinkedList<Dtran>>();
+	NewNode startNode;
+	NewNode nowNode;
+	NewNode lastNode;
+	MatchAbout ma;
+	String rex;
+	String content;
+	char[] chList;
+	int pointer=0;
+	String tran;
+	Stack<Recall> stack = new Stack<Recall>();
+	public Match(String str){
+		this.rex=str;
+		MainClass mc= new MainClass(str);
+		ToDFA td=new ToDFA(mc.getNodeList(),mc.getInputChar());
+		this.dtrans=td.getDtrans();
+		this.startNode=td.getStartNewNode();
+		this.nowNode=startNode;
+		ManagerK.handle(td.getNewNodes(),mc.getNodeList());
+		ma= new MatchAbout(ManagerK.getEn());
+		//System.out.println(startNode.getId());
+	}
+	public void match(String content){
+		this.content=content;
+		chList=content.toCharArray();
+	}
+	public boolean find(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.